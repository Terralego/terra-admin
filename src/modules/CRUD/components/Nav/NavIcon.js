import React from 'react';

const NavIcon = ({ src, alt = '', ...rest }) => src && (
  <span className="bp3-icon" {...rest}>
    <img className="CRUD-nav__icon" src={src} alt={alt} />
  </span>
);

export default NavIcon;
