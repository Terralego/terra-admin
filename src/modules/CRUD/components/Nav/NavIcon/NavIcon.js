import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from '@blueprintjs/core';

const NavIcon = ({ src, alt = '', isGroup, className = '', ...rest }) =>
  (src
    ? (
      <span {...rest} className={`${className} bp3-icon`}>
        <img className="CRUD-nav__icon" src={src} alt={alt} />
      </span>
    )
    : <Icon {...rest} className={className} icon={isGroup ? 'layers' : 'layer'} color="#333" />
  );

export default NavIcon;

NavIcon.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  isGroup: PropTypes.bool,
};

NavIcon.defaultProps = {
  src: undefined,
  alt: undefined,
  isGroup: false,
};
