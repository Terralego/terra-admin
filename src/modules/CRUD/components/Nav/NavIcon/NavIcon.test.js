import React from 'react';
import renderer from 'react-test-renderer';

import NavIcon from './NavIcon';

jest.mock('@blueprintjs/core', () => ({
  Icon: ({ icon }) => <div>{icon}</div>,
}));

it('should snapshot correctly when src is filled', () => {
  const tree = renderer.create((
    <NavIcon src="path/to/image.png" />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should snapshot correctly one layer Icon without src filled', () => {
  const tree = renderer.create((
    <NavIcon />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should snapshot correctly layers Icon without src filled', () => {
  const tree = renderer.create((
    <NavIcon isGroup />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
