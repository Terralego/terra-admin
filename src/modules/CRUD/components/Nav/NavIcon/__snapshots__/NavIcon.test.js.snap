// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot correctly layers Icon without src filled 1`] = `
<div>
  layers
</div>
`;

exports[`should snapshot correctly one layer Icon without src filled 1`] = `
<div>
  layer
</div>
`;

exports[`should snapshot correctly when src is filled 1`] = `
<span
  className=" bp3-icon"
>
  <img
    alt=""
    className="CRUD-nav__icon"
    src="path/to/image.png"
  />
</span>
`;
